src/pykeen/utils.py

```diff
@@ -917,6 +930,43 @@ def extend_batch(
     return hrt_batch
 
 
+def check_shapes(
+    *x: Tuple[Union[torch.Tensor, Tuple[int, ...]], str],
+    raise_on_errors: bool = True,
+) -> bool:
+    """Verify that a sequence of tensors are of matching shapes.
+
+    :param x:
+        A tuple (tensor, shape), where tensor is a tensor, and shape is a string, where each character corresponds to
+        a (named) dimension. If the shapes of different tensors share a character, the corresponding dimensions are
+        expected to be of equal size.
+    :param raise_on_errors:
+        Whether to raise an exception in case of a mismatch.
+
+    :return:
+        Whether the shapes matched.
+
+    :raises ValueError:
+        If the shapes mismatch and raise_on_error is True.
+    """
+    dims: Dict[str, Tuple[int, ...]] = dict()
+    errors = []
+    for actual_shape, shape in x:
+        if isinstance(actual_shape, torch.Tensor):
```

Comment:
use `torch.is_tensor`?

Reply:
nope, `torch.is_tensor()` breaks mypy (this code is exactly what's in torch.is_tensor though)